### PR TITLE
Add support for VRF leaking under router-general

### LINKED ISSUE #669: Add support for VRF leaking under router general
**Issue Type**

- [ ] Template enhancement
- [x] Role enhancement
- [ ] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**

For a specific project, we need to activate VRF leaking under `router general` which is not supported yet by __arista.avd.eos_cli_config_gen__

**Describe the solution you'd like**

Implement doc and eos template for `router general` with following data model:

```yaml
router_general:
  vrfs:
    < destination vrf >:
      leak:
        source_vrf: < source vrf > 
        policy: < route-map filtering >
```

### PR DESCRIPTION
**Pull request by titom73**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes issue #669 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Add support for `router_general` for vrf leaking

```yaml
router_general:
  vrfs:
    < destination-vrf >:
      leak_routes:
        - source_vrf: < source-vrf >
          subscribe_policy: < route-map policy >
        - source_vrf: < source-vrf >
          subscribe_policy: < route-map policy >
```

## How to test

Please check Molecule scenario

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

**Comment by titom73** [comment]
Add support for multiple source VRFs

```
+router general
+   vrf test
+      leak routes source-vrf test3 subscribe-policy all
+      leak routes source-vrf test2 subscribe-policy all
+!
```

**Comment by titom73** [comment]
Ready for testing and review.
cc: @sebastianw @mpergament

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 1]
```diff
@@ -1449,6 +1450,19 @@ ip_routing: < true | false >
 ipv6_unicast_routing: < true | false >
 ```
 
+#### Router General configuration
+
+```yaml
+router_general:
+  vrfs:
+    < destination-vrf >:
+      leaks:
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: I would recommend updating `leaks:` to `leak_routes:` to match CLI.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 2]
```diff
@@ -1449,6 +1450,19 @@ ip_routing: < true | false >
...
+        - source_vrf: < source-vrf >
+          policy: < route-map policy >
+        - source_vrf: < source-vrf >
+          policy: < route-map policy >
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: I would recommend updating `policy:` to `subscribe_policy:` to match CLI.

**Review by carlbuchmann — CHANGES_REQUESTED** [review]
I recommend some changes to the data model to more closely match the CLI.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 1 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Fixed in 7b15495

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 2 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Fixed in 7b15495

**Comment by titom73** [comment]
Changes applied in commit 7b15495.
Ready for final review and merge

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Pull request merged**